### PR TITLE
Port clap from yml to structopt

### PR DESCRIPTION
**Pull request by tedil**
As the title says, this PR ports the CLI from [clap](https://github.com/clap-rs/clap) yaml to [structopt](https://docs.rs/structopt/0.3.21/structopt/index.html).

### CODE REVIEW

File: src/fastq/collapse_reads_to_fragments/mod.rs [hunk 1]
```diff
@@ -110,9 +110,17 @@ pub fn call_consensus_reads_from_paths(
...
+                "Reading input files:\n    {}\n    {}",
+                fq1.as_ref().display(),
+                fq2.as_ref().display()
+            );
+            info!(
+                "Writing output to:\n    {}\n    {}",
+                fq1_out.as_ref().display(),
+                fq2_out.as_ref().display()
+            );
+            match (fq1.as_ref().ends_with(".gz"), fq2.as_ref().ends_with(".gz"), fq1_out.as_ref().ends_with(".gz"), fq2_out.as_ref().ends_with(".gz")) {
```
Comment (tedil) [inline src/fastq/collapse_reads_to_fragments/mod.rs:123]: It should be possible to get rid of this whole match statement by either boxing the readers and writers or with conditionally initializing uninitialized variables. I'll post a suggestion "tomorrow".

Comment (fxwiegand) [reply src/fastq/collapse_reads_to_fragments/mod.rs:123]: Does this really belong to this PR though? I mean we could easily put this in here but this should be somehow related to porting clap to structopt right?

**Comment by fxwiegand** [comment]
Some tests for `collapse-reads-to-fragments` still fail with the following error:
```
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.33.3/src/app/usage.rs:475:22
```
I am not sure whether this is my fault or some issue with structopt or clap. Any idea @tedil?

**Comment by natir** [comment]
We can maybe wait the release of clap 3.0.0, this release integrate structopt directly, with same functionality and almost same api.

**Comment by tedil** [comment]
> We can maybe wait the release of clap 3.0.0, this release integrate structopt directly, with same functionality and almost same api.
> 
> clap 3.0.0-beta2 is actually available.

Just had a quick look at the changelogs / information for switching to clap 3.0 and it seems like that would only require us to replace all occurrences of `structopt` with `clap`. I'd like *not* to use beta versions if it can be avoided, and since the changes that need to be done once clap 3.0 is released officially, there's not much point in waiting, I think

**Review by johanneskoester — APPROVED** [review]
Thanks a lot, this is much appreciated!

**Pull request merged**